### ISSUE TITLE
Still connection issues in 0.2.2

### ISSUE THREAD

**Issue by sarahmva** [georgezhao2010/midea_ac_lan#30]
Unfortunately, I am still seeing the connection become available/unavailable all the time. My system is a V3 system ( 'version': 3)

Here is the debug log.
```
2022-07-26 18:33:45 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Connected
2022-07-26 18:33:45 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Handshaking
2022-07-26 18:33:45 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Connected
2022-07-26 18:33:45 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Handshaking
2022-07-26 18:33:46 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Authentication success
2022-07-26 18:33:46 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Status update: {'available': True}
2022-07-26 18:33:46 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Sending: {'header': 'aa23ac8f000000000003', 'payload': '418100ff03ff00020000000000000000000000000000005d', 'message type': '0x3', 'body type': '0x41'}
2022-07-26 18:33:46 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Authentication success
2022-07-26 18:33:46 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Status update: {'available': True}
2022-07-26 18:33:46 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Sending: {'header': 'aa23ac8f000000000003', 'payload': '418100ff03ff00020000000000000000000000000000005f', 'message type': '0x3', 'body type': '0x41'}
2022-07-26 18:33:56 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] does not supports the protocol MessageQuery, has been ignored
2022-07-26 18:33:56 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Sending: {'header': 'aa14acb8000000000003', 'payload': 'b1034200180015005e', 'message type': '0x3', 'body type': '0xb1'}
2022-07-26 18:33:56 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] does not supports the protocol MessageQuery, has been ignored
2022-07-26 18:33:56 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Sending: {'header': 'aa14acb8000000000003', 'payload': 'b10342001800150060', 'message type': '0x3', 'body type': '0xb1'}
2022-07-26 18:34:06 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] does not supports the protocol MessageNewProtocolQuery, has been ignored
2022-07-26 18:34:06 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Refresh status is all timed out
2022-07-26 18:34:06 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Status update: {'available': False}
2022-07-26 18:34:06 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] does not supports the protocol MessageNewProtocolQuery, has been ignored
2022-07-26 18:34:06 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Refresh status is all timed out
2022-07-26 18:34:06 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Status update: {'available': False}
2022-07-26 18:34:11 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Connected
2022-07-26 18:34:11 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Handshaking
2022-07-26 18:34:11 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Connected
2022-07-26 18:34:11 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Handshaking
2022-07-26 18:34:11 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Authentication success
2022-07-26 18:34:11 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Status update: {'available': True}
2022-07-26 18:34:11 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Sending: {'header': 'aa23ac8f000000000003', 'payload': '418100ff03ff000200000000000000000000000000000061', 'message type': '0x3', 'body type': '0x41'}
2022-07-26 18:34:11 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Authentication success
2022-07-26 18:34:11 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Status update: {'available': True}
2022-07-26 18:34:11 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Sending: {'header': 'aa23ac8f000000000003', 'payload': '418100ff03ff000200000000000000000000000000000063', 'message type': '0x3', 'body type': '0x41'}
2022-07-26 18:34:21 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] does not supports the protocol MessageQuery, has been ignored
2022-07-26 18:34:21 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Sending: {'header': 'aa14acb8000000000003', 'payload': 'b10342001800150062', 'message type': '0x3', 'body type': '0xb1'}
2022-07-26 18:34:21 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] does not supports the protocol MessageQuery, has been ignored
2022-07-26 18:34:21 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Sending: {'header': 'aa14acb8000000000003', 'payload': 'b10342001800150064', 'message type': '0x3', 'body type': '0xb1'}
2022-07-26 18:34:31 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] does not supports the protocol MessageNewProtocolQuery, has been ignored
2022-07-26 18:34:31 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Refresh status is all timed out
2022-07-26 18:34:31 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Status update: {'available': False}
2022-07-26 18:34:31 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] does not supports the protocol MessageNewProtocolQuery, has been ignored
2022-07-26 18:34:31 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Refresh status is all timed out
2022-07-26 18:34:31 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Status update: {'available': False}
2022-07-26 18:34:36 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Connected
2022-07-26 18:34:36 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Handshaking
2022-07-26 18:34:36 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Connected
2022-07-26 18:34:36 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Handshaking
2022-07-26 18:34:37 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Authentication success
2022-07-26 18:34:37 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Status update: {'available': True}
2022-07-26 18:34:37 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Sending: {'header': 'aa23ac8f000000000003', 'payload': '418100ff03ff000200000000000000000000000000000065', 'message type': '0x3', 'body type': '0x41'}
2022-07-26 18:34:37 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Authentication success
2022-07-26 18:34:37 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Status update: {'available': True}
2022-07-26 18:34:37 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Sending: {'header': 'aa23ac8f000000000003', 'payload': '418100ff03ff000200000000000000000000000000000067', 'message type': '0x3', 'body type': '0x41'}
2022-07-26 18:34:47 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] does not supports the protocol MessageQuery, has been ignored
2022-07-26 18:34:47 DEBUG (Thread-9) [custom_components.midea_ac_lan.midea.core.device] [3***************6] Sending: {'header': 'aa14acb8000000000003', 'payload': 'b10342001800150066', 'message type': '0x3', 'body type': '0xb1'}
2022-07-26 18:34:47 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] does not supports the protocol MessageQuery, has been ignored
2022-07-26 18:34:47 DEBUG (Thread-8) [custom_components.midea_ac_lan.midea.core.device] [3***************3] Sending: {'header': 'aa14acb8000000000003', 'payload': 'b10342001800150068', 'message type': '0x3', 'body type': '0xb1'}
```

**Comment by georgezhao2010** [comment]
> Unfortunately, I am still seeing the connection become available/unavailable all the time. My system is a V3 system ( 'version': 3)
> 
> Here is the debug log.
> 

It's work normally in 0.1.19?

**Comment by sarahmva** [comment]
> > Unfortunately, I am still seeing the connection become available/unavailable all the time. My system is a V3 system ( 'version': 3)
> > Here is the debug log.
> 
> It's work normally in 0.1.19?

Yes, I downgraded to 0.1.19 and everything is back to normal. No disconnects.

**Comment by nc1120917035** [comment]
me too!

**Comment by georgezhao2010** [comment]
> > > Unfortunately, I am still seeing the connection become available/unavailable all the time. My system is a V3 system ( 'version': 3)
> > > Here is the debug log.
> > 
> > 
> > It's work normally in 0.1.19?
> 
> Yes, I downgraded to 0.1.19 and everything is back to normal. No disconnects.

@sarahmva @nc1120917035 

- Upgrade to 0.2.2
- Download attached file and unzip it
- Put `message.py` to `custome_component/midea_ac_lan/midea/devices/ac`, overwrite the old one.
- Restart HA 

See if problem is solve and let me know.

[message.zip](https://github.com/georgezhao2010/midea_ac_lan/files/9200709/message.zip)

**Comment by sarahmva** [comment]
Thank you very much. Yes, this fixes the connection Problem.

**Comment by georgezhao2010** [comment]
> Thank you very much. Yes, this fixes the connection Problem.

OK, I'll fix it in v0.2.3

could you star this repository if it's helpful to you.
